packages/fluentui/react-builder/src/ability-attributes/ErrorReporter.ts

```diff
@@ -0,0 +1,276 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+/* eslint-disable */
+
+import {
+  AbilityAttributesError,
+  ATTRIBUTE_NAME_ERROR_ID,
+  ATTRIBUTE_NAME_ERROR_MESSAGE,
+  ErrorReporter as ErrorReporterBase,
+  WindowWithDevEnv,
+} from './DevEnvTypes';
+
+interface ErrorMetadata {
+  id: string;
+  element: ValidatedHTMLElement;
+  error: AbilityAttributesError;
+  visible: boolean;
+  shouldRerender?: boolean;
+  rendered?: HTMLDivElement;
+  locator?: string;
+}
+
+interface ValidatedHTMLElement extends HTMLElement {
+  __aaError?: ErrorMetadata;
+}
+
+let _lastId = 0;
+let _errors: { [id: string]: ErrorMetadata } = {};
+
+export class ErrorReporter extends ErrorReporterBase {
```

Comment:
What is the difference between this class and the `errorReporter` object that you are passing into the `setup` method in the `AbilityAttributesValidator.tsx` file?

It looks like in `AbilityAttributesValidator.tsx` you pass the object that has a `report` callback which is called when an error is found. Presumably you are then going to take that information and render a report using React from where the `AbilityAttributesValidator` component currently returns `null`? That looks like a great approach!

What I don's understand is what the `ErrorReporter` class is doing and why it is generating it's own HTML. :)